### ISSUE TITLE
[Bug]: [JSON] No values in output_fields when searching specifying JSON fields

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#24132]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master-20230515-30415e1b
- Deployment mode(standalone or cluster): both
- MQ type(rocksmq, pulsar or kafka):  all
- SDK version(e.g. pymilvus v2.0.0rc2):pymilvus 2.4.0.dev27
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

 No values in output_fields when searching specifying JSON fields
```
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "int64 >= 0 && int64 <= 5", output_fields=["json_field"])
>>> res[0][0]
id: 0, distance: 0.0002716142334975302, entity: {}
```

### Expected Behavior

Return json fields in output_fields

### Steps To Reproduce

```python
from pymilvus import CollectionSchema, FieldSchema
from pymilvus import Collection
from pymilvus import connections
from pymilvus import DataType
from pymilvus import Partition
from pymilvus import utility

connections.connect()
dim = 128
int64_field = FieldSchema(name="int64", dtype=DataType.INT64, is_primary=True)
float_field = FieldSchema(name="float", dtype=DataType.FLOAT)
bool_field = FieldSchema(name="bool", dtype=DataType.BOOL)
string_field = FieldSchema(name="string", dtype=DataType.VARCHAR, max_length=65535)
json_field = FieldSchema(name="json_field", dtype=DataType.JSON)
float_vector = FieldSchema(name="float_vector", dtype=DataType.FLOAT_VECTOR, dim=dim)
schema = CollectionSchema(fields=[int64_field, float_field, bool_field, json_field, float_vector])
collection = Collection("test_search_collection_binbin_tmp", schema=schema)
import random
nb=3000
vectors = [[random.random() for _ in range(dim)] for _ in range(nb)]
import numpy as np
res = collection.insert([[i for i in range(nb)], [np.float32(i) for i in range(nb)], [np.bool_(i) for i in range(nb)], [{"number": i, "string": "milvus", "bool": True, "list": [j for j in range(0, i)]} for i in range(0, nb)], vectors])
index_param = {"index_type": "IVF_SQ8", "metric_type": "L2", "params": {"nlist": 8192}}
collection.flush()
collection.create_index("float_vector", index_param, index_name="index_name_1")
collection.load()
default_search_params = {"metric_type": "L2", "params": {"nprobe": 10}}
limit = 10
nq =1
res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "int64 >= 0 && int64 <= 5", output_fields=["json_field"])
```

**Comment by yah01** [comment]
/assign @binbinlv
fixed with latest pymilvus 2.4.0.dev30

**Comment by binbinlv** [comment]
Verified and Fixed:
pymilvus: 2.4.0.dev30
milvus: master-20230516-ccd68501

```
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "int64 >= 0 && int64 <= 5", output_fields=["json_field"])
id: 0, distance: 0.00032137916423380375, entity: {'json_field': b'{"number":0,"string":"milvus","bool":true,"list":[]}'}
```

**Comment by binbinlv** [comment]
This exists in 2.2.* branch too.

**Comment by binbinlv** [comment]
Reopen for waiting for 2.2.* branch code merge.

**Comment by binbinlv** [comment]
Verified and fixed in 2.2.* branch too:
milvus: [2.2.0-20230520-1eb804fc](https://hub.docker.com/layers/milvusdb/milvus/2.2.0-20230520-1eb804fc/images/sha256-f1fce0b4ba76324119d89cebe860886197da341061e6688f0ef9da34719ef485?context=explore)
pymilvus: 2.2.9.dev12

```
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "int64 >= 0 && int64 <= 5", output_fields=["json_field"])
>>> res[0][0]
id: 0, distance: 0.0, entity: {'json_field': b'{"number":0,"string":"milvus","bool":true,"list":[]}'}
>>> res[0][1]
id: 3, distance: 19.007768630981445, entity: {'json_field': b'{"number":3,"string":"milvus","bool":true,"list":[0,1,2]}'}
```
